File: lib/data/models/commit_model.dart
import '../../domain/entities/commit.dart' as entity;
import '../../domain/entities/enums.dart';
import '../database/app_database.dart';

/// 提交数据模型 - Entity ↔ Data 转换
extension CommitDataExtension on CommitData {
  /// 转换为领域实体
  entity.Commit toEntity() {
    return entity.Commit(
      id: id,
      taskId: taskId,
      branchId: branchId,
      message: message,
      type: CommitType.fromValue(type.value),
      createdAt: createdAt,
    );
  }
}

/// 提交实体扩展 - 转换为 drift 数据
extension CommitEntityExtension on entity.Commit {
  /// 转换为 drift 数据对象
  CommitData toData() {
    return CommitData(
      id: id,
      taskId: taskId,
      branchId: branchId,
      message: message,
      type: type,
      createdAt: createdAt,
    );
  }
}

File: web/src/data/models/commit-model.ts
import { CommitType } from '../../domain/entities/enums';
import { Commit } from '../../domain/entities/commit';

export interface CommitRecord {
  id: string;
  taskId: string;
  branchId: string;
  message: string;
  type: number;
  createdAt: string;
}

export const CommitModel = {
  toRecord(entity: Commit): CommitRecord {
    return {
      id: entity.id,
      taskId: entity.taskId,
      branchId: entity.branchId,
      message: entity.message,
      type: entity.type,
      createdAt: entity.createdAt.toISOString(),
    };
  },

  toEntity(record: CommitRecord): Commit {
    return new Commit({
      id: record.id,
      taskId: record.taskId,
      branchId: record.branchId,
      message: record.message,
      type: CommitType.fromValue(record.type),
      createdAt: new Date(record.createdAt),
    });
  },
};
